include/opendht/value.h

```diff
@@ -132,7 +132,7 @@ struct Value
     public:
         static Filter chain(Filter&& f1, Filter&& f2) {
             return [f1,f2](const Value& v){
-                return f1(v) && f2(v);
+                return (f1 ? f1(v) : true) && (f2 ? f2(v) : true);
```

Comment:
for better performance this could be simplified: chain could just return f1 or f2 if the other is not set (same for bellow).